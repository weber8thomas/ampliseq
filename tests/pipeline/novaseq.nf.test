nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_novaseq"
    tag "pipeline"

    test("Paired-End Novaseq") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/software_versions.yml").exists() },
                { assert snapshot(path("$outputDir/dada2/ASV_seqs.fasta"),
                                path("$outputDir/dada2/ASV_table.tsv"),
                                path("$outputDir/dada2/DADA2_stats.tsv"),
                                path("$outputDir/dada2/DADA2_table.rds"),
                                path("$outputDir/dada2/DADA2_table.tsv")).match("dada2") },
                { assert new File("$outputDir/fastqc/S1_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/S1_2_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/S2_1_fastqc.html").exists() },
                { assert new File("$outputDir/fastqc/S2_2_fastqc.html").exists() },
                { assert snapshot(path("$outputDir/input/Samplesheet_novaseq.tsv")).match("input") },
                { assert snapshot(path("$outputDir/multiqc/multiqc_data/multiqc_fastqc.txt"),
                                path("$outputDir/multiqc/multiqc_data/multiqc_general_stats.txt")).match("multiqc") },
                { assert new File("$outputDir/summary_report/summary_report.html").exists() }
            )
        }
    }
}
