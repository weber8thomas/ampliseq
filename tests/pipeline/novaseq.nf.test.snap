{
    "input": {
        "content": [
            "Samplesheet_novaseq.tsv:md5,bcca78fe24736f4ee4ea26fdbd47ee64"
        ],
        "timestamp": "2023-06-20T00:10:02+0000"
    },
    "dada2": {
        "content": [
            "ASV_seqs.fasta:md5,60c7cc4669df672109fce853dfa874fd",
            "ASV_table.tsv:md5,db74c8b5abf372fdaed19d0bf23c91c8",
            "DADA2_stats.tsv:md5,5f82ca12036dae5da84a93b6fce96c36",
            "DADA2_table.rds:md5,e7cc351e115cccdeabd0d0e9e7d9045a",
            "DADA2_table.tsv:md5,fba33455974dd6c07ce54a72f7147425"
        ],
        "timestamp": "2023-06-20T00:10:02+0000"
    },
    "multiqc": {
        "content": [
            "multiqc_fastqc.txt:md5,a99548124efd185a7d4e33245edcb81b",
            "multiqc_general_stats.txt:md5,195ce3e0c1b032459040613802d25d50"
        ],
        "timestamp": "2023-06-20T00:10:02+0000"
    }
}
